servers/src/main/java/tachyon/master/journal/JsonJournalFormatter.java

```
@@ -19,402 +19,64 @@
+    return new ObjectMapper().configure(JsonGenerator.Feature.AUTO_CLOSE_TARGET, false)
+            .configure(SerializationFeature.CLOSE_CLOSEABLE, false);
   }
 
+  public static final ObjectMapper OBJECT_MAPPER = createObjectMapper();
+  public static final ObjectWriter OBJECT_WRITER = OBJECT_MAPPER.writer();
+
   @Override
   public void serialize(SerializableJournalEntry entry, OutputStream outputStream)
       throws IOException {
-    // serialize parameters
-    Map<String, Object> parameters = entry.getParameters();
-    JsonEntry jsonEntry = new JsonEntry(entry.getSequenceNumber(), entry.getType());
-    for (Map.Entry<String, Object> parameter : parameters.entrySet()) {
-      jsonEntry.withParameter(parameter.getKey(), parameter.getValue());
-    }
-    writeEntry(jsonEntry, outputStream);
+    writeEntry(entry, outputStream);
   }
 
-  private void writeEntry(JsonEntry entry, OutputStream os) throws IOException {
-    JsonEntry.OBJECT_WRITER.writeValue(os, entry);
+  private void writeEntry(SerializableJournalEntry entry, OutputStream os) throws IOException {
+    OBJECT_WRITER.writeValue(os, entry);
     (new DataOutputStream(os)).write('\n');
   }
 
   @Override
   public JournalInputStream deserialize(final InputStream inputStream) throws IOException {
     return new JournalInputStream() {
-      private JsonParser mParser = JsonEntry.createObjectMapper().getFactory()
-          .createParser(inputStream);
+      private JsonParser mParser = OBJECT_MAPPER.getFactory().createParser(inputStream);
       private long mLatestSequenceNumber = 0;
 
       @Override
       public JournalEntry getNextEntry() throws IOException {
-        JsonEntry entry;
+        JsonNode entryNode;
         try {
-          entry = mParser.readValueAs(JsonEntry.class);
+          entryNode = mParser.readValueAs(JsonNode.class);
         } catch (JsonProcessingException e) {
           return null;
```

Shall we fail fast if a exception is encountered?